modin/numpy/arr.py

```diff
@@ -0,0 +1,333 @@
+            result = result.replace(numpy.inf, 0)
+        return array(_query_compiler=result, _ndim=new_ndim)
+
+    __floordiv__ = _floor_divide
+
+    def _power(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        if caller != self:
+            # No workaround possible until broadcasting fixed. GH#5529.
+            pass
+        result = caller._query_compiler.pow(callee._query_compiler, **kwargs)
+        return array(_query_compiler=result, _ndim=new_ndim)
+
+    __pow__ = _power
+
+    def _prod(self, axis=None, out=None, keepdims=None, where=None):
+        if axis is None:
+            result = self._query_compiler.prod(axis=0).prod(axis=1)
+            return array(_query_compiler=result)
+        else:
+            result = self._query_compiler.prod(axis=axis)
+            return array(_query_compiler=result)
+
+    def _multiply(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        result = caller._query_compiler.mul(callee._query_compiler, **kwargs)
+        return array(_query_compiler=result, _ndim=new_ndim)
+
+    __mul__ = _multiply
+
+    def _remainder(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        if caller != self:
+            # No workaround possible until broadcasting fixed. GH#5529.
+            pass
+        result = caller._query_compiler.mod(callee._query_compiler, **kwargs)
+        if any(callee._query_compiler.eq(0).to_pandas()):
+            result = result.replace(numpy.NaN, 0)
+        return array(_query_compiler=result, _ndim=new_ndim)
+
+    __mod__ = _remainder
+
+    def _subtract(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        result = caller._query_compiler.sub(callee._query_compiler, **kwargs)
+        if caller != self:
+            result = result.rsub(0)
+        return array(_query_compiler=result, _ndim=new_ndim)
+
+    __sub__ = _subtract
+
+    def _sum(
+        self, axis=None, dtype=None, out=None, keepdims=None, initial=None, where=None
+    ):
+        result = self._query_compiler.sum(axis=axis)
+        if dtype is not None:
+            result = result.astype(dtype)
+        if out is not None:
+            out._query_compiler = result
+            return
+        if axis is None:
+            return
```

Comment:
this feels weird, so it's not doing an in-place add but returns nothing?.. looks like a bug